templates/server.conf.erb

```diff
@@ -125,7 +125,9 @@ push "<%= @redirect_gateway %>"
 push "dhcp-option DNS <%= @dns %>"
 <% end -%>
 
-<% if @domain != '' -%>
+<% if @domain.is_a? Array -%>
+<% @domain.each do |d| -%>push "dhcp-option DOMAIN <%= "#{d}\"\n" %><% end %>
```

Comment:
We can probably do `Array(@domain).each ...` here and skip the logic to figure out its data type.  The explicit newline here shouldn't be necessary.  Was there a missing newline in your testing?